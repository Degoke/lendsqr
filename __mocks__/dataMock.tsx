import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
