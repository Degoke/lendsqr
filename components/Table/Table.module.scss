.table {
  padding: 1rem;
  width: 100%;

  th,
  td {
    padding: 0.5rem;
    border-bottom: 1px solid #213f7d1a;
    text-align: left;
    font-size: 14px;
  }

  th {
    font-size: 12px;
    padding: 0.5rem;

    span {
      margin-right: 0.5rem;
    }
  }
}

.container {
  position: relative;
  overflow-x: scroll;
}
