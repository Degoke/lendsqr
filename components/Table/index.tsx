import { IUser } from '../../utils/types/user'
import FilterMenu from '../FilterMenu'
import TableHeader from '../TableHeader'
import TableRow from '../TableRow'
import styles from './Table.module.scss'

type Props = {
  users: IUser[]
}

export default function Table({ users }: Props) {
  return (
    <div className={styles.container}>
      <table className={styles.table}>
        <thead>
          <tr>
            <TableHeader text="organization" />
            <TableHeader text="username" />
            <TableHeader text="email" />
            <TableHeader text="phonenumber" />
            <TableHeader text="date joined" />
            <TableHeader text="status" />
            <th></th>
          </tr>
        </thead>
        <tbody>
          {users?.map((user) => (
            <TableRow key={user.id} user={user} />
          ))}
        </tbody>
      </table>
      <FilterMenu isOpen={false} />
    </div>
  )
}
