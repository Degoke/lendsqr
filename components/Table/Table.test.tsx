import { render, screen } from '@testing-library/react'
import Table from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('Table tests', () => {
  it('renders Table component unchanged', () => {
    const { container } = render(
      <Table users={[]} toggleFilterMenu={jest.fn} />
    )
    expect(container).toMatchSnapshot()
  })
})
