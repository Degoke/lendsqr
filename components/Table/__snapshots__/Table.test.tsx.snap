// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table tests renders Table component unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th
            class="tableHeader"
          >
            <span>
              organization
            </span>
            <img
              alt="filter"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style="color: transparent;"
              width="12"
            />
          </th>
          <th
            class="tableHeader"
          >
            <span>
              username
            </span>
            <img
              alt="filter"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style="color: transparent;"
              width="12"
            />
          </th>
          <th
            class="tableHeader"
          >
            <span>
              email
            </span>
            <img
              alt="filter"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style="color: transparent;"
              width="12"
            />
          </th>
          <th
            class="tableHeader"
          >
            <span>
              phonenumber
            </span>
            <img
              alt="filter"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style="color: transparent;"
              width="12"
            />
          </th>
          <th
            class="tableHeader"
          >
            <span>
              date joined
            </span>
            <img
              alt="filter"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style="color: transparent;"
              width="12"
            />
          </th>
          <th
            class="tableHeader"
          >
            <span>
              status
            </span>
            <img
              alt="filter"
              data-nimg="1"
              decoding="async"
              height="12"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style="color: transparent;"
              width="12"
            />
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
    <div
      class="popover popover"
    />
  </div>
</div>
`;
